internal/receiver/discoveryreceiver/evaluator.go

```diff
@@ -116,79 +116,30 @@ func (e *evaluator) evaluateMatch(match Match, pattern string, status discovery.
-			// value set by the initial receiver config parser.
-			from.PutStr(discovery.ReceiverConfigAttr, receiverAttrs[discovery.ReceiverConfigAttr])
-			hasTemporaryReceiverConfigAttr = true
-		}
-	}
-
-	from.Range(func(k string, v pcommon.Value) bool {
-		if k == discovery.ReceiverConfigAttr && e.config.EmbedReceiverConfig {
-			configVal := v.AsString()
-			updatedWithObserverAttr := fmt.Sprintf("%s.%s", receiverUpdatedConfigAttr, observerID)
-			if updatedConfig, ok := receiverAttrs[updatedWithObserverAttr]; ok {
-				configVal = updatedConfig
-			} else if observerID != "" {
-				var err error
-				if updatedConfig, err = addObserverToEncodedConfig(configVal, observerID); err != nil {
-					// log failure and continue with existing config sans observer
-					e.logger.Debug(fmt.Sprintf("failed adding %q to %s", observerID, discovery.ReceiverConfigAttr), zap.String("receiver.type", receiverType), zap.String("receiver.name", receiverName), zap.Error(err))
-				} else {
-					e.logger.Debug("Adding watch_observer to embedded receiver config receiver attrs", zap.String("observer", corr.observerID.String()), zap.String("receiver.type", receiverType), zap.String("receiver.name", receiverName))
-					e.correlations.UpdateAttrs(corr.receiverID, map[string]string{
-						updatedWithObserverAttr: updatedConfig,
-					})
-					configVal = updatedConfig
-				}
-			}
-			v = pcommon.NewValueStr(configVal)
+		embeddedConfig := map[string]any{}
+		rEntry := cfg.Receivers[corr.receiverID] // it's safe to assume this exists.
+		embeddedReceiversConfig := map[string]any{}
+		receiverConfig := map[string]any{}
+		receiverConfig["rule"] = rEntry.Rule
+		receiverConfig["config"] = rEntry.Config
+		receiverConfig["resource_attributes"] = rEntry.ResourceAttributes
+		embeddedReceiversConfig[corr.receiverID.String()] = receiverConfig
+		embeddedConfig["receivers"] = embeddedReceiversConfig
+		if observerID != "" && observerID != discovery.NoType.String() {
+			embeddedConfig["watch_observers"] = []string{observerID}
 		}
-		if _, ok := to.Get(k); !ok {
-			v.CopyTo(to.PutEmpty(k))
+		cfgYaml, err := yaml.Marshal(embeddedConfig)
+		if err != nil {
+			e.logger.Error("failed embedding receiver config", zap.String("observer", observerID), zap.Error(err))
 		}
-		return true
-	})
-	if hasTemporaryReceiverConfigAttr {
-		from.Remove(discovery.ReceiverConfigAttr)
-	}
-}
-
-func addObserverToEncodedConfig(encoded, observerID string) (string, error) {
-	cfg := map[string]any{}
-	dBytes, err := base64.StdEncoding.DecodeString(encoded)
-	if err != nil {
-		return "", err
-	}
-	if err = yaml.Unmarshal(dBytes, &cfg); err != nil {
-		return "", err
-	}
-
-	cfg["watch_observers"] = []string{observerID}
-
-	var cfgYaml []byte
-	if cfgYaml, err = yaml.Marshal(cfg); err != nil {
-		return "", fmt.Errorf("failed embedding receiver config to include %q: %w", observerID, err)
+		to.PutStr(discovery.ReceiverConfigAttr, base64.StdEncoding.EncodeToString(cfgYaml))
```

Comment:
Why don't we have to update the correlation's attribute value here, like we used to when this logic was in `receiverCreatorReceiversConfig`?

Reply:
Because it's not being used anymore

Reply:
It'll eventually go to attributes once we stop emitting entities here. But I need to make some more changes before that